addons/mail/static/src/messaging/entity_field.js

```diff
@@ -342,7 +340,7 @@ class Field {
                     for (const otherEntity of entity[currentFieldName]) {
                         const field = otherEntity.constructor.fields[relatedFieldName];
                         if (field && field.hashes.includes(hash)) {
-                            this._registerToCompute(otherEntity, field);
+                            entity.env.entityManager.registerToComputeField(otherEntity, field);
```

Comment:
I think the ideal path to reach env should be
field -> Entity (class) -> entityManager -> env
entity (instance) -> Entity (class) -> entityManager -> env

Source of truth for env should be the only thing that is stored directly in the env, so the manager. All others should have getter directing towards that.

It does no make sense to store env on individual fields/entities (either class or instance) because we cannot expect them to work if extracted outside of the whole tree here. (as opposed to for example components that can be tested without a parent)